### PR TITLE
Ongoing OIDC: add IsPrimaryAuthMethod bool to AuthMethods returned from repos

### PR DESCRIPTION
**Pull request by jimlambrt**
This is only at the repo level.  We still need to plumb this into the API, SDK and CLI

### CODE REVIEW

**Review by talanknight — COMMENTED** [review]
Looks good.

File: internal/auth/oidc/repository_auth_method_read_test.go [hunk 1]
```diff
@@ -96,6 +100,9 @@ func TestRepository_LookupAuthMethod(t *testing.T) {
 			}
 			require.NoError(err)
 			assert.EqualValues(tt.want, got)
+			if got != nil {
+				assert.Equal(tt.wantIsPrimary, got.IsPrimaryAuthMethod)
```
[the remark this answers is not in the document]
Comment (jimlambrt) [reply internal/auth/oidc/repository_auth_method_read_test.go:104]: ty!

File: internal/auth/password/options.go [hunk 2]
```diff
@@ -81,3 +82,10 @@ func WithConfiguration(config Configuration) Option {
 		o.withConfig = config
 	}
 }
+
+// WithOrder provides an optional with order clause.
```
[the remark this answers is not in the document]
Comment (jimlambrt) [reply internal/auth/password/options.go:86]: ty!

File: internal/auth/password/repository_authmethod.go [hunk 3]
```diff
@@ -268,3 +269,76 @@ func (r *Repository) UpdateAuthMethod(ctx context.Context, authMethod *AuthMetho
...
+	case len(ams) > 1:
+		return nil, errors.New(errors.NotSpecificIntegrity, op, fmt.Sprintf("%s matched more than 1 ", authMethodId))
+	default:
+		return ams[0], nil
+	}
+}
+
+// getAuthMethods allows the caller to either lookup a specific AuthMethod via
+// its id or search for a set AuthMethods within a set of scopes.  Passing both
+// scopeIds and a authMethodId is an error. The WithLimit option is supported
```
[the remark this answers is not in the document]
Comment (jimlambrt) [reply internal/auth/password/repository_authmethod.go:293]: ty!

**Pull request merged**